File: medianScores.py
import bisect
def medianScores(scores):
    ans = [scores[0]]
    tmp = [scores[0]]
    median = None

    for i in range(1, len(scores)):
        bisect.insort(tmp, scores[i])

        if len(tmp) % 2 == 0:
            median = (tmp[math.ceil(len(tmp) / 2)] +
                      tmp[math.ceil(len(tmp) / 2) - 1]) / 2
        else:
            median = tmp[math.ceil(len(tmp) / 2) - 1]

        ans.append(math.ceil(median))
    return ans

File: medianScores.js
function medianScores(scores) {
    let ans = [scores[0]],
        tmp = [scores[0]],
        median;

    for (let i = 1; i < scores.length; i++) {
        inOrder(tmp, scores[i]);

        if (tmp.length % 2 !== 0) {
            median = tmp[Math.round(tmp.length / 2) - 1];
        } else {
            median = (tmp[tmp.length / 2] + tmp[tmp.length / 2 - 1]) / 2;
        }

        ans.push(Math.round(median));
    }

    return ans;
}

function inOrder(arr, item) {
    /* Insert item into arr keeping low to high order */

    let ix = 0;
    while (ix < arr.length) {
        if (item < arr[ix]) { break; }
        ix++;
    }

    arr.splice(ix, 0, item);
    return arr
}